EIPS/eip-3026.md

```diff
@@ -0,0 +1,287 @@
+- Input has invalid length
+- Field element encoding rules apply (obviously)
+- Scalar encoding rules apply (obviously)
+
+##### ABI for G1 multiexponentiation
+
+G1 multiplication call expects `256*k` bytes as an input that is interpreted as the byte concatenation of `k` slices, each of them being a byte concatenation of the point-encoding of a G1 point (`192` bytes) and the encoding of a scalar value (`64` bytes). Output is an encoding of the multiexponentiation operation result.
+
+Error cases:
+- Any of the G1 points being not on the curve
+- Input has invalid length
+- Field element encoding rules apply (obviously)
+- Scalar encoding rules apply (obviously)
+
+##### ABI for G2 addition
+
+G2 addition call expects `384` bytes as an input that is interpreted as the byte concatenation of two G2 points (point-encoded as `192` bytes each). Output is a point-encoding of the addition operation result.
+
+Error cases:
+- Either of points being not on the curve
+- Input has invalid length
+- Field elements encoding rules apply (obviously)
+
+##### ABI for G2 multiplication
+G2 multiplication call expects `256` bytes as an input that is interpreted as the byte concatenation of the point-encoding of a G2 point (`192` bytes) and the encoding of a scalar value (`64` bytes). Output is an encoding of multiplication operation result.
+
+Error cases:
+- Point being not on the curve must result in error
+- Field elements encoding rules apply (obviously)
+- Input has invalid length
+
+##### ABI for G2 multiexponentiation
+
+G2 multiplication call expects `240*k` bytes as an input that is interpreted as byte concatenation of `k` slices each of them being a byte concatenation of encoding of G2 point (`192` bytes) and encoding of a scalar value (`48` bytes). Output is an encoding of multiexponentiation operation result.
+
+Error cases:
+- Any of G2 points being not on the curve must result in error
+- Field elements encoding rules apply (obviously)
+- Input has invalid length
+
+##### ABI for pairing
+
+Pairing call expects `384*k` bytes as an input, that is interpreted as the byte concatenation of `k` slices. Each slice has the following structure:
+- `192` bytes G1 point encoding
+- `192` bytes G2 point encoding
+
+Output is `32` bytes representing a boolean:
+
+- `0x0000000000000000000000000000000000000000000000000000000000000001` if the pairing result is equal the to multiplicative identity in the pairing target field; and
+- `0x0000000000000000000000000000000000000000000000000000000000000000` otherwise.
+
+Error cases:
+- Any of the G1 or G2 points being not on the curve
+- Any of the G1 or G2 points being not in the correct subgroup
+- Input has invalid length
+- Field elements encoding rules apply (obviously)
+
+#### Prevention of DDoS on error handling
+
+This precompile performs extensive computations and in case of any errors during execution it MUST consume all gas from the the gas schedule for the corresponding operation.
```

Comment:
```suggestion
This precompile performs extensive computations and in case of any errors during execution it **MUST** consume all gas from the the gas schedule for the corresponding operation.
```